beater/server.go

```diff
@@ -35,11 +37,22 @@ func run(server *http.Server, config Config) error {
 
 	ssl := config.SSL
 	if ssl.isEnabled() {
-		return server.ListenAndServeTLS(ssl.Cert, ssl.PrivateKey)
+		logp.Info("Loading server certificate: %s", config.SSL.Certificate.Certificate)
+		logp.Info("Loading private key: %s", config.SSL.Certificate.Key)
+		cert, err := outputs.LoadCertificate(&config.SSL.Certificate)
+		if err != nil {
+			return err
```

Comment:
This code path should also be tested, I think

Reply:
@jalvz While I'm writing a test for this path, I saw that we are completely hiding error from the `setupServer` function (`server_test` file). Particularly on the `go run` statement.

If we check the error while running the server, we could remove all the `checkErrMsg`tests.

For instance, the `TestServerSecureBadDomain` test, would only have to check if an error has occured with the following msg : `http: TLS handshake error from 127.0.0.1:60368: remote error: tls: bad certificate`.
Again, for the `TestServerBadProtocol` test, we would only have to check for an error like `http: TLS handshake error from 127.0.0.1:60386: tls: oversized record received with length 21536` which tells us that we are trying to reach the https server with the http protocol.

Do you agree ?

Reply:
@sterchelen that sounds good to me, but how do check the error while running the server?